### DISCUSSION TITLE
Window_Activated event replacment missing?

### DISCUSSION THREAD

**Discussion by sboulema** [VsixCommunity/Community.VisualStudio.Toolkit#87] [Q&A]
I found Toolkit replacement for all my oldskool DTE events, except for the Window_Activated event?

Any tips on how to implement it? The current set of Window events doesn't seem to match the old behaviour and is giving me errors when used with VS2022.

Thanks in advance!

Samir

**Comment by madskristensen** [comment]
In the latest _-pre_ versions of the package, you can subscribe to `VS.Events.WindowEvents.ActiveFrameChanged` and other events like it. Those events are meant to replace the DTE versions of the same

**Reply by sboulema** [reply]
Thanks for the quick reply!

If I use that event I get the following exception :(

```
System.InvalidCastException: 'Specified cast is not valid.'
   at Microsoft.VisualStudio.Shell.Interop.IVsUIShell7.AdviseWindowFrameEvents(IVsWindowFrameEvents eventSink)
   at Community.VisualStudio.Toolkit.Events.get_WindowEvents()
```

My code:

```
...
VS.Events.WindowEvents.ActiveFrameChanged += WindowEvents_ActiveFrameChanged;
...

private void WindowEvents_ActiveFrameChanged(object sender, ActiveFrameChangeEventArgs e)
{
    // Magic should happen here
}
```

**Reply by madskristensen** [reply]
I can't repro the issue on the latest version of the toolkit source code. That eventhandler has also changed signature. Can you try the latest CI build of the NuGet package? To do that, create a file called **nuget.config** in your solution folder, and paste in the following content:

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="nuget.org" value="https://www.nuget.org/api/v2/" />
    <add key="Toolkit CI" value="https://ci.appveyor.com/nuget/community-visualstudio-toolkit" />
  </packageSources>
</configuration>
```

Are you by chance using version 16.0 of the toolkit, but running it inside VS 2022?

**Reply by sboulema** [reply]
Running the 15.0 version of the toolkit inside VS 2022.

Does this mean I finally should properly split the VSIX into different versions?

**Reply by madskristensen** [reply]
That might be the case. You can easily test it by temporarily upgrade the toolkit package to 17.x and see if that fixes it

**Reply by sboulema** [reply]
Upgrading to 17.0.75.165-pre fixes my problem!

**Reply by madskristensen** [reply]
If the issue comes back when downgrading to **15**.0.75.165-pre, then a split is needed

**Reply by sboulema** [reply]
Checked and yes a split is needed.

Fun project for the weekend :D

Thanks for all the help!
